app/demo-payment/payment-failed.controller.js

```diff
@@ -3,18 +3,19 @@
 const urlJoin = require('url-join')
 
 const staticify = require('../../config/server').staticify
-const { SELFSERVICE_TRANSACTIONS_URL, DOCS_URL } = require('../../config')
+const { SELFSERVICE_DEMO_PAYMENT_RETURN_URL, DOCS_URL } = require('../../config')
 
 const CONFIRM_FAILURE_VIEW = './demo-payment-failed'
 
 const data = {
   scenarioDocsLink: urlJoin(DOCS_URL, '/#payment-flow-payment-fails'),
   cardNumbersDocsLink: urlJoin(DOCS_URL, '/#mock-card-numbers-for-testing-purposes'),
-  transactionsLink: SELFSERVICE_TRANSACTIONS_URL,
   exampleImgSrc: staticify.getVersionedPath('/images/error-page.png')
 }
 
 module.exports = (req, res) => {
+  const transactionsLink = SELFSERVICE_DEMO_PAYMENT_RETURN_URL.replace(':productExternalId', req.params.productExternalId)
+  data.transactionsLink = transactionsLink
```

Comment:
As we're setting something in the global `data` object that is not a constant, I think we should just initialise `data` within the function rather than as a global constant so there isn't any risk of unexpected behaviour if the state is left over.